File: java/tests/Boza.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <sstream>
#include <new>
#include <stdlib.h>

using namespace std;
    
void nested(int nRows,int nElementsInRow,int startingNumber){
  
  for (int i = startingNumber ; i < nRows+startingNumber ; i++ ){
    for (int j = i ; j < nElementsInRow + i ; j++){
      cout << j << " " ;
    }
    cout << endl;
  }
}


    
int main (){
  
  cout << " Enter number of rows " << endl;
  int n_rows = 0;
  cin >> n_rows;
  cout << " Enter total elements in row " << endl;
  int n_elements_in_row = 0;
  cin >> n_elements_in_row ;
  cout << " Enter starting number " << endl;
  int start = 0;
  cin >> start ;
  

  nested(n_rows,n_elements_in_row,start);

}

File: java/tests/Boza.java
public class Boza {
    
    
    public static void nested(int nRows,int nElementsInRow,int startingNumber){	
	for (int i = startingNumber ; i < nRows+startingNumber ; i++ ){
	    for (int j = i ; j < nElementsInRow + i ; j++){
		System.out.print(j + " ");
	    }
	    System.out.println();
	}
    }
        
    public static void nested_rec(int nr,int nel,int start){
	
	for (int i = start; i < nel+start ; i++){
	    if (nr == 0) break;
	    else {
		System.out.print(i + " ");
		if (i+1 == nel+start){
		    nr = nr-1;
		    System.out.println();
		    nested_rec(nr,nel,start+1);
		}
	    }
	}
	
    }
    
    public static void main (String[] args){

	nested_rec(5,4,2);

    }
}
